pkg/specgen/generate/storage.go

```diff
@@ -81,14 +81,38 @@ func finalizeMounts(ctx context.Context, s *specgen.SpecGenerator, rt *libpod.Ru
 	}
 
 	for _, v := range s.Volumes {
-		if err = parse.ValidateVolumeCtrDir(v.Dest); err != nil {
-			return nil, nil, nil, err
+		// check if named volume has to be overlayed
+		overlayFlag := false
```

Comment:
Should this be a shared function?  Or is this only called here?

Reply:
@rhatdan apart from this, check happens in specgen but not at any other place. Initially i wanted to do this when specgen creation happens but we cannot transform named volume to overlay in specgen since we need access to runtime for mountPoint.

So after specgen creation this is the second place  i dont think any another place has this check.